### ISSUE TITLE
New Design

### ISSUE THREAD

**Issue by Sandared** [osgi-users-virtual/osgi-users-virtual.github.io#6]
Hi Neal, 

after our short discussion on Twitter (@NewsOSGi) about the VUG, I thought of how we could attract more people to the forum/VUG aside from publishing more posts.

What do you think about another design and of using [Docusaurus](https://docusaurus.io/en/) instead of Jekyll?

Reasons for Docusaurus:
* easier setup for future contributor (however, that's just my personal opinion after working with Liquid templates and jekyll for some time ;) )
* good responsive design which has an up to date look and feel 

If you think this is a sensible idea, then I would start to develop an initial setup and then create a PR.

Kind regards,
Thomas

**Comment by rotty3000** [comment]
No offense intended at all to the initial site template chosen by @njbartlett. It was already a significant effort that got us going...

I tend to agree we could do better particularly if someone is willing to do the work! I'd personally love to see a new site template :)

**Comment by Sandared** [comment]
Hi @rotty3000 , that's nice to hear. However, I would want to wait for @njbartlett 's opinion before I start working on this. I neither want to wast my nor your time if such a (radical) change is not wished for by all parties ;)

Kind regards,
Thomas

**Comment by njbartlett** [comment]
If you are confident that you can produce something better, by all means go ahead! I am not married to the current design.

On the other hand I'm not entirely sure why the alternative would be easier to setup. A lot of people are already familiar with making GitHub PRs and authoring in Markdown. Can you describe what advantages Docusaurus has in terms of ease-of-use?

Regarding the template, this is constrained by two factors:

1. Royalty free and can be used with minimal attribution (I don't want a huge "Site Design by XXX" on every page).
2. Not looking exactly the same as every other Bootstrap-based site on the internet.

**Comment by Sandared** [comment]
Hi Neil,
I didn't intend to change the setup regarding GitHub pages, only to swap the static site generator. I've done quite some work in the past with Jekyll and it's not a pleasant experience (Liquidtemplates are horrible :/)

Docusaurus is just another static site generator that gets you started really easy (npm + yarn and you are reday to go) has a reasonable standard design and is quite fast (Jekyll got slow if you used a recursive tree like template)

Regarding licences: I'm not sure what I have to watch out for. Docusaurus itself is MIT licensed. I guess it's the same for their standard theme which we can customize.

Regarding Bootstrap-based: I'm not sure what types of design that would exclude ;) If you could give me examples of what would be ok for you, then I could try to keep the design similar to them. If you want to see what can be made with Docusaurus, then you can have alook here https://docusaurus.io/en/users
These are all websites that are made with Docusaurus.

Kind regards,
Thomas

**Comment by njbartlett** [comment]
> Docusaurus is just another static site generator that gets you started really easy (npm + yarn and you are reday to go) has a reasonable standard design and is quite fast (Jekyll got slow if you used a recursive tree like template)

Okay but the great advantage of Jekyll is that the generator is built into GitHub. So authors only have to check-in sources, e.g. MD files.

Is Docusaurus also supported by GitHub? Or would authors have to run the generator locally and check-in the generated outputs?
 
> Regarding licences: I'm not sure what I have to watch out for. Docusaurus itself is MIT licensed. I guess it's the same for their standard theme which we can customize.

The license of the generator is not as important as the license of the template. So long as we don't have to pay a royalty or put a massive "Site Design By..." banner on the site, it should be fine.

> 
> Regarding Bootstrap-based: I'm not sure what types of design that would exclude ;) If you could give me examples of what would be ok for you, then I could try to keep the design similar to them. If you want to see what can be made with Docusaurus, then you can have alook here https://docusaurus.io/en/users
> These are all websites that are made with Docusaurus.

This is not what I meant.

When Bootstrap was first released by Twitter, a large number of sites were built using its default template. As a result, they all looked the same and as a user you could instantly tell when a site was using Bootstrap. I just don't want the OSGi VUG site to look identical to all the other sites generated by Docusaurus (or any other template/generator).

> 
> Kind regards,
> Thomas

**Comment by Sandared** [comment]
> Okay but the great advantage of Jekyll is that the generator is built into GitHub. So authors only have to check-in sources, e.g. MD files.
> 
> Is Docusaurus also supported by GitHub? Or would authors have to run the generator locally and check-in the generated outputs?

I don't think so. The user would have to generate the site via `npm run build` and then push the result. I didn't know that Jekyll is completely integrated into Github. Sorry. If you want to stick with Jekyll then we could consider the template I once created for enRoute: https://sandared.github.io/documentation_template/ 
You can easily change the color scheme if wished. The rest (layout and stuff) is a little bit more work to do. Is this template too much like everyone else? Or would that be ok for you?

> The license of the generator is not as important as the license of the template. So long as we don't have to pay a royalty or put a massive "Site Design By..." banner on the site, it should be fine.

If the MIT licence doesn't force you to put such a "Site Design By" anywhere then we're safe with the Docusaurus standard theme I think.

**Comment by njbartlett** [comment]
> I don't think so. The user would have to generate the site via `npm run build` and then push the result. I didn't know that Jekyll is completely integrated into Github. Sorry. If you want to stick with Jekyll then we could consider the template I once created for enRoute: https://sandared.github.io/documentation_template/
> You can easily change the color scheme if wished. The rest (layout and stuff) is a little bit more work to do. Is this template too much like everyone else? Or would that be ok for you?

Yes I think I would prefer to stick with Jekyll until we can find a better solution. If we ever do migrate to something else then I would hope the Markdown content would transfer over seamlessly.

That template looks great! Yes some tinkering with colours would be good, perhaps to fit in more with the OSGi orange-and-blue theme?

Another part of the current site I like is the FAQ which is assembled from individual files. My aim here was to allow FAQs to be easily added by contributors.

> If the MIT licence doesn't force you to put such a "Site Design By" anywhere then we're safe with the Docusaurus standard theme I think.

Yes that sounds safe enough.

**Comment by Sandared** [comment]
>Yes I think I would prefer to stick with Jekyll until we can find a better solution. If we ever do migrate to something else then I would hope the Markdown content would transfer over seamlessly.

Ok then we stick with Jekyll

> That template looks great! Yes some tinkering with colours would be good, perhaps to fit in more with the OSGi orange-and-blue theme?

Thanks! Nice to hear. That color scheme would be no problem. I even did one version for enRoute that had the OSGi colors but they didn't really want it. Maybe I promoted it the wrong way ;)

> Another part of the current site I like is the FAQ which is assembled from individual files. My aim here was to allow FAQs to be easily added by contributors.

That should be no problem too. It will take some time however. maybe I can copy your code and fit it into the design of my template.

If the template is ok for you I would start to create a first version over the next weeks or so?

Kind regards,
Thomas

**Comment by Sandared** [comment]
I've created a first version of the new design here: https://sandared.github.io/osgi-users-virtual.github.io
I've copied the blog contents of the VUG. The rest is still placeholder stuff and the documentation of the template itself. 

WDYT?

Any absolute no gos? Any general comments? WDYT?

**Comment by rotty3000** [comment]
My first thought is that there is too much happening. All those things need justification. I would take a few away so we don't feel so guilty about not having content there.

Suggestions:
- remove the front page section with 3 horizontal blocks (starting with `Beautiful landing page`)
- replace news on front page with blogs summaries
- remove all but the narrow bottom part of the footer
- would you be willing to grant the copyright to the VUG itself?

**Comment by rotty3000** [comment]
But to be clear... it does look really good! :+1:

**Comment by Sandared** [comment]
Hi Ray,

thanks! I changed the following:
* removed parts of footer
* removed peptalks
* changed news to be blogposts (this we could enhance to encompass not only blogposts but all relevant changes (FAQs, Resources) and it should be generated automatically)
* added forum, but this does not work yet. maybe @njbartlett could provide some advice here?

And yes the vug can have the copyright. As I already mentioned, I created this template once for enRoute and sadly it wasn't accepted, so I'm glad if it is used by someone else instead of being only wasted time ;)

Kind regards,
Thomas

**Comment by Sandared** [comment]
I've just added the FAQ section too. You now can add FAQs by adding a new .md file into the faqs folder. 
I think only Forum (which might start working once deployed to this repository? @njbartlett ?) and the Join page are missing now.

Any comments on the current state? Or any change requests :) ?

**Comment by Sandared** [comment]
@rotty3000 @njbartlett : any comments/wishes/etc. ? :)

**Comment by Sandared** [comment]
Hey guys, I know you might be quite busy right now, but I've only time to work in this issue from 22.03. - 01.04. After that I will be on my well deserved vacation ;)  So if you could review what I've done so far and tell me what needs to be done in order to get this whole issue finished then this would be great :)

Kind regards,
Thomas

**Comment by rotty3000** [comment]
Can you send a PR for us to review?

**Comment by Sandared** [comment]
Hi @rotty3000 , I made a PR yesterday but didn't check that you've made changes before. Sry, I'm currently in the final stage of my PhD exams and my brain seems to be elsewhere. If you just want to see how the current draft looks you can also go to  https://sandared.github.io/osgi-users-virtual.github.io and have a look until I fix the PR next week or so.

**Comment by Sandared** [comment]
@rotty3000 and @njbartlett hopefully this time I did the PR right ;)

It would be great if you could give me feedback on this PR soon as I'm on vacation from 08.04. and I would love to see this work finished before ;)

Kind regards,
Thomas

**Comment by Sandared** [comment]
@rotty3000 / @njbartlett : I have only a few days left before going on vacation, so any feedback on this issue is highly appreciated.
